cmd/version.go

```diff
@@ -11,23 +12,50 @@ import (
 	apiversion "github.com/redhat-certification/chart-verifier/pkg/chartverifier/version"
 )
 
+// Print version and commit ID as json blob
+var asData bool
+
 func init() {
 	rootCmd.AddCommand(NewVersionCmd())
 }
 
+// CommitIDLong contains the commit ID the binary was build on. It is populated at build time by ldflags.
+// If you're running from a local debugger it will show an empty commit ID.
+var CommitIDLong string
+
+type VersionContext struct {
+	Version string `json:"version"`
+	Commit  string `json:"commit"`
+}
+
+var Version = VersionContext{
+	Version: apiversion.GetVersion(),
+	Commit:  CommitIDLong,
+}
+
 func NewVersionCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "version",
 		Short: "print the chart-verifier version information",
+		Long:  "git commit information for this particular binary build is included at build time and can be accessed by this command",
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return runVersion(os.Stdout)
 		},
 	}
+	cmd.Flags().BoolVar(&asData, "as-data", false, "output the version and commit ID information in JSON format")
 	return cmd
 }
 
 func runVersion(out io.Writer) error {
-	fmt.Fprintf(out, "v%s\n", apiversion.GetVersion())
+	if asData {
+		marshalledVersion, err := json.Marshal(Version)
+		if err != nil {
+			return err
```

Comment:
Instead of returning an error here, what do you think about falling back to the regular way of reporting the version? Maybe with an extra trailing statement saying something like "request for data format failed"?

Reply:
Fair point there. LGTM